segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/NonAppendExtendedS3IntegrationTest.java

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server.host;
+
+import com.emc.object.s3.S3Config;
+import com.emc.object.s3.jersey.S3JerseyClient;
+import com.google.common.base.Preconditions;
+import io.pravega.segmentstore.server.store.ServiceBuilder;
+import io.pravega.segmentstore.server.store.ServiceBuilderConfig;
+import io.pravega.segmentstore.storage.AsyncStorageWrapper;
+import io.pravega.segmentstore.storage.SimpleStorageFactory;
+import io.pravega.segmentstore.storage.Storage;
+import io.pravega.segmentstore.storage.StorageFactory;
+import io.pravega.segmentstore.storage.chunklayer.ChunkedSegmentStorage;
+import io.pravega.segmentstore.storage.chunklayer.ChunkedSegmentStorageConfig;
+import io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperConfig;
+import io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperLogFactory;
+import io.pravega.segmentstore.storage.metadata.ChunkMetadataStore;
+import io.pravega.segmentstore.storage.rolling.RollingStorage;
+import io.pravega.storage.extendeds3.ExtendedS3ChunkStorage;
+import io.pravega.storage.extendeds3.ExtendedS3Storage;
+import io.pravega.storage.extendeds3.ExtendedS3StorageConfig;
+import io.pravega.storage.extendeds3.S3ClientMock;
+import io.pravega.storage.extendeds3.S3Mock;
+import io.pravega.test.common.TestUtils;
+import lombok.Getter;
+import lombok.val;
+import org.junit.After;
+import org.junit.Before;
+
+import java.net.URI;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * End-to-end tests for SegmentStore, with integrated Extended S3 Storage and DurableDataLog.
+ */
+public class NonAppendExtendedS3IntegrationTest extends BookKeeperIntegrationTestBase {
+    //region Test Configuration and Setup
+
+    private String s3ConfigUri;
+    private S3Mock s3Mock;
+
+    /**
+     * Starts BookKeeper.
+     */
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        s3ConfigUri = "http://127.0.0.1:" + TestUtils.getAvailableListenPort() + "?identity=x&secretKey=x";
+        s3Mock = new S3Mock();
+        this.configBuilder.include(ExtendedS3StorageConfig.builder()
+                .with(ExtendedS3StorageConfig.CONFIGURI, s3ConfigUri)
+                .with(ExtendedS3StorageConfig.BUCKET, "kanpravegatest"));
+    }
+
+    @Override
+    @After
+    public void tearDown() throws Exception {
```

Comment:
This is not necessary. JUnit will see this method from the parent class and execute it anyway.